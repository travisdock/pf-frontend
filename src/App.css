#home2 {
  height:100vh;
  display: flex;
  flex-direction: column;
}
div {
  font-family: 'Roboto Condensed', sans-serif;
  max-width: 100%;
  max-height: 100%;
}

button {
  font-family: 'Roboto Condensed', sans-serif;
}

.ReactTable .-pagination .-btn {
  margin: 0px !important;
}

pre {
  margin: 0px;
}

textarea, input {
  font-family: 'Roboto Condensed', sans-serif;
}

.ReactTable .-pagination .-pageJump input {
  margin: 0px;
}

div.about {
  padding: 0px 20px;
}

h1.logo {
  font-family: 'Roboto Condensed', sans-serif;
  font-size: 55px;
  margin: auto;
}

div a.active {
  background-color: #3A5199;
  color: #E2E2E2;
}

.navbar {
  display: flex;
  width: 100%;
}

.navbar .navlinks {
  width: 100%;
}

.navbar a {
  float: left;
  color: #3F4D4E;
  text-align: center;
  padding: 12px 14px;
  text-decoration: none;
  font-size: 17px;
  border-radius: 20px;
  margin: 10px 4px;
}

.navbar a:hover {
    background-color: #ddd;
    color: black;
}

.navbar .more {
  float: right;
  background-color: #DEDCE4;
  border: 2px solid #ccc;
  border-radius: 5px;
  padding: 12px 14px;
  text-decoration: none;
  font-size: 15px;
  margin: 10px 4px 10px 0px;
}

.content {
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: 100%;
}

div.ui {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-items: center;
  flex-direction: column;
}

.content .input_form {
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
}


.content .chart-content {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0px 7px;
  flex-flow: wrap;
  overflow-y: scroll;
}

div form {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-direction: column;
  max-width: 100%;
}

select {
  font-size: 16px;
  padding: 12px 20px;
  margin: 8px 0;
  border: 2px solid #ccc;
}
textarea, input {
  padding: 12px 20px;
  margin: 8px 5px;
  border: 2px solid #ccc;
  font-size: 16px;
  border-radius: 4px;
  resize: vertical;
  max-width: 80%;
}

.checkboxes {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 8px;
}

.checkbox_wrapper {
  display: flex;
}

.content button {
  border: 2px solid #ccc;
  background-color: #DEDCE4;
  margin: 3px 3px;
  font-size: 16px;
  padding: 12px 20px;
  border-radius: 5px;
  box-shadow: 0 2px 3px 0 rgba(0,0,0,0.2), 0 2px 3px 0 rgba(0,0,0,0.19);
}

div.table-popup {
  display: flex;
  /* justify-content: space-around; */
  align-items: center;
  flex-direction: column;
  border: 3px solid #000406;
  padding: 12px 5px;
  border-radius: 6px;
  background-color: white;
}
div.navbar-popup {
  display: flex;
  /* justify-content: space-around; */
  align-items: center;
  flex-direction: column;
  border: 3px solid #000406;
  padding: 12px 5px;
  border-radius: 6px;
  background-color: white;
}

@media only screen and (max-width: 500px) {
    div.content div.popup-content {
      width: 80% !important;
    }
}

div.content div.popup-content {
  padding: 0px !important;
  border: none !important;
  border-radius: 6px;
  font-size: 21px;
  background-color: transparent !important;
}

div.content div.table-content {
  width: 100%;
  margin: 0 8px 0 8px;
  height: 100%;
  display: flex;
}

#table {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.ReactTable {
  height: 100%;
}


.bb-title {
  font-size: 40px;
}


input[type="date"],
input[type="number"],
input[type="password"],
input[type="text"],
select:focus,
textarea {
  font-size: 16px;
}

.import_container {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 15px;
}
.img_container {
   width: 100%;
}
.img_container img {
   display: block;
   width: 100%;
   max-width: 448px;
   height: auto;
   margin: auto;
}