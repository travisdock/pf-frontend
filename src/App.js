import React, { Component } from 'react';
import './App.css';

import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import { applyMiddleware, compose, createStore } from 'redux';
import { Route, Switch } from 'react-router';
import { ConnectedRouter, connectRouter, routerMiddleware } from 'connected-react-router'
import reducer from './reducers/index';

import Navbar from './components/navbar'
import Login from './containers/login'
import Signup from './containers/signup'

const history = createBrowserHistory()
const initialState = {}

const store = createStore(
  connectRouter(history)(reducer),
  initialState,
  compose(
    applyMiddleware(
      routerMiddleware(history)
    )
  )
);

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <div>
            <Navbar />
            <Switch>
              <Route exact path="/login" render={() => <Login />} />
              <Route exact path="/about" render={() => (<div>Match about</div>)} />
              <Route exact path="/dashboard" render={() => (<div>Match dash</div>)} />
              <Route exact path="/signup" render={() => <Signup />} />
              <Route path="/" render={() => <Login />} />
            </Switch>
          </div>
        </ConnectedRouter>
      </Provider>
    );
  }
}

export default App;
