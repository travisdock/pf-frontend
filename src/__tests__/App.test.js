import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { App } from '../App';

jest.mock('../containers/Dashboard', () => () => 'Dashboard')
jest.mock('../containers/Home', () => () => 'Home')
jest.mock('../components/Navbar', () => () => 'Navbar')

// Mock fetch
function mockFetch(data) {
    return jest.fn(() =>
      Promise.resolve({
        ok: true,
        json: () => data
      })
    );
  }

const userProps = {
    current_user: {user_id: 1},
    load: false,
    login: jest.fn(),
    logout: jest.fn()
};
const loadProps = {
    current_user: {user_id: null},
    load: true,
    login: jest.fn(),
    logout: jest.fn()
}
const noUserProps = {
    current_user: {user_id: null},
    load: false,
    login: jest.fn(),
    logout: jest.fn()
}

describe('Snapshots', () => {
    it('renders a <App/> snapshot with no user', () => {
        const component = renderer.create(<App {...noUserProps} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a <App/> snapshot with user, dashboard', () => {
        const component = renderer.create(
            <MemoryRouter initialEntries={[ '/dashboard' ]}>
                <App {...userProps} />
            </MemoryRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a <App/> snapshot with user, home', () => {
        const component = renderer.create(
            <MemoryRouter initialEntries={[ '/' ]}>
                <App {...userProps} />
            </MemoryRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a <App/> snapshot with load', () => {
        const component = renderer.create(
            <MemoryRouter initialEntries={[ 'app.com/login' ]}>
                <App {...loadProps} />
            </MemoryRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Routes correctly', () => {
    it('logs in returning user', async () => {
        window.fetch = mockFetch('success')
        jest.spyOn(window.localStorage.__proto__, 'getItem')
            .mockImplementation(() => 1)

        const wrapper = await mount(
            <MemoryRouter initialEntries={[ 'app.com/login' ]}>
              <App {...userProps} />
            </MemoryRouter>
        );
        await wrapper.update();

        expect(userProps.login).toHaveBeenCalledWith('success');
    });

    it('logs out returning user', async () => {
        window.fetch = mockFetch({error: 'error'})
        jest.spyOn(window.localStorage.__proto__, 'getItem')
            .mockImplementation(() => 1)

        const wrapper = await mount(
            <MemoryRouter initialEntries={[ 'app.com/login' ]}>
              <App {...userProps} />
            </MemoryRouter>
        );
        await wrapper.update();

        expect(userProps.logout).toHaveBeenCalled();
    });
 
    it('logs out by default', async () => {
        // jest.spyOn(window.localStorage.__proto__, 'getItem')
        //     .mockImplementation(() => null)

        const wrapper = await mount(
            <MemoryRouter initialEntries={[ 'app.com/login' ]}>
              <App {...userProps} />
            </MemoryRouter>
        );
        await wrapper.update();

        expect(userProps.logout).toHaveBeenCalled();
    });
});
