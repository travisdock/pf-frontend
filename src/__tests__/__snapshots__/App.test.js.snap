// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders a <App/> snapshot with load 1`] = `
<div
  id="home"
>
  <div
    id="home2"
  >
    Navbar
  </div>
</div>
`;

exports[`Snapshots renders a <App/> snapshot with no user 1`] = `
<div
  id="home"
/>
`;

exports[`Snapshots renders a <App/> snapshot with user, dashboard 1`] = `
<div
  id="home"
>
  <div
    id="home2"
  >
    Navbar
    Dashboard
  </div>
</div>
`;

exports[`Snapshots renders a <App/> snapshot with user, home 1`] = `
<div
  id="home"
>
  <div
    id="home2"
  >
    Navbar
    Home
  </div>
</div>
`;
